ocfweb/docs/views/servers.py

```diff
@@ -57,81 +56,90 @@ def has_munin(self):
+    result = {}
+    for d in output:
+        hostname = d['certname'].split('.')[0]
+        result[hostname] = d['value']
+    return result
+
+
 @periodic(120)
 def get_hosts():
-    return [
-        Host.from_ldap(
-            hostname='hal',
-            type='hypervisor',
-            children=[
-                Host.from_ldap(hostname)
-                for hostname in (
-                    'fallingrocks',
-                    'monsoon',
-                )
-            ],
-        ),
-
-        Host.from_ldap(
-            hostname='jaws',
-            type='hypervisor',
-            children=[
-                Host.from_ldap(hostname)
-                for hostname in (
-                    'anthrax',
-                    'biohazard',
-                    'death',
-                    'dementors',
-                    'democracy',
-                    'firestorm',
-                    'flood',
-                    'lightning',
-                    'maelstrom',
-                    'pestilence',
-                    'reaper',
-                    'supernova',
-                    'thunder',
-                    'tsunami',
-                    'werewolves',
-                    'whirlwind',
-                    'whiteout',
-                    'zombies',
-                )
-            ],
-        ),
-
-        Host.from_ldap(
-            hostname='pandemic',
-            type='hypervisor',
-            children=[
-                Host.from_ldap(hostname)
-                for hostname in (
-                    'pileup',
-                )
-            ],
-        ),
-
-        Host.from_ldap(
-            hostname='riptide',
-            type='hypervisor',
-        ),
-
-        Host.from_ldap(
-            hostname='corruption',
-            type='server',
-        ),
-
-        Host('blackhole', 'network', 'Managed Cisco Catalyst 2960S-48TS-L Switch.', []),
-
-        Host('logjam', 'printer', '', []),
-        Host('pagefault', 'printer', '', []),
-        Host('papercut', 'printer', '', []),
-        Host.from_ldap('overheat', type='raspi'),
-        Host.from_ldap('tornado', type='nuc'),
-
-        *(Host.from_ldap(desktop, type='desktop')
-          for desktop in sorted(list_desktops())),
-    ]
+    def create_hosts(lst):
+        """Accepts a list of raw ldap output, returns a dictionary of host
+        objects indexed by hostname
+        """
+        hosts = {}
+        for h in lst:
+            if not is_hidden(h):
+                description = h.get('description', [''])[0]
+                hostname = h['cn'][0]
+                hosts[hostname] = (Host(hostname, h['type'], description, ()))
+        return hosts
+
+    servers = create_hosts(hosts_by_filter('(type=server)'))
+    desktops = create_hosts(hosts_by_filter('(type=desktop)'))
+    misc = create_hosts(hosts_by_filter('(type=printer)'))
+    hypervisors = {}
+
+    physical_hosts = query_puppet(TEMP_IS_VIRT_FALSE)
```

Comment:
I've found that you can directly get a list of hypervisors with the PQL query `resources[certname] { type = "Class" and title = "Ocf_kvm" }`. (Note that the capital O is not a typo.)

What this query does is that it checks whether a node has included the `ocf_kvm` Puppet class, which is the class that makes a given server a hypervisor. Nodes include `ocf_kvm` based on their configuration in hieradata. For example, [here](https://github.com/ocf/puppet/blob/33e2e00a32cd6f3d44940a6a0b6cf89e0e280c49/hieradata/nodes/jaws.yaml#L2) is the specific item that tells `jaws` to include `ocf_kvm` and thus make itself a hypervisor. If you have some free time, I encourage you to browse other files in the same directory to see which other nodes have that class included and which ones do not.